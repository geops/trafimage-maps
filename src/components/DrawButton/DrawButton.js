import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { RiPencilFill } from 'react-icons/ri';
import IconButton from '@material-ui/core/IconButton';

function DrawButton({ children, ...buttonProps }) {
  const mapsetUrl = useSelector((state) => state.app.mapsetUrl);
  const { t } = useTranslation();

  return (
    <IconButton
      href={`${mapsetUrl}?parent=${encodeURIComponent(window.location)}`}
      target="blank_"
      title={t('Zeichnen')}
      size="medium"
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...buttonProps}
    >
      {children || <RiPencilFill focusable={false} />}
    </IconButton>
  );
}

DrawButton.propTypes = {
  children: PropTypes.element,
};

DrawButton.defaultProps = {
  children: null,
};

export default React.memo(DrawButton);
