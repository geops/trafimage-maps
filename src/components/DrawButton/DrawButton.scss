.ta-draw-icon {
  svg {
    font-size: 21px;
    height: 100%;
  }
}
