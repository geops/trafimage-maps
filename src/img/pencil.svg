<svg xmlns="http://www.w3.org/2000/svg" width="17.29" height="17.29" viewBox="0 0 17.29 17.29" stroke="currentColor" fill="currentColor"><path d="M0,13.688v3.6H3.6L14.224,6.668l-3.6-3.6ZM17.009,3.883a.956.956,0,0,0,0-1.354L14.762.281a.956.956,0,0,0-1.354,0L11.65,2.038l3.6,3.6Z"/></svg>