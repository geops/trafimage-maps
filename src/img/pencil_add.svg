<svg xmlns="http://www.w3.org/2000/svg" width="17.998" height="17.29" viewBox="0 0 17.998 17.29" stroke="currentColor" fill="currentColor">
  <g id="Gruppe_121" data-name="Gruppe 121" transform="translate(-279.96 -1680)">
    <path id="Icon_material-edit" data-name="Icon material-edit" d="M0,13.688v3.6H3.6L14.224,6.668l-3.6-3.6ZM17.009,3.883a.956.956,0,0,0,0-1.354L14.762.281a.956.956,0,0,0-1.354,0L11.65,2.038l3.6,3.6Z" transform="translate(279.96 1680)"/>
    <path id="Pfad_118" data-name="Pfad 118" d="M318.387,1684.6v6.895" transform="translate(-23.876 5.766)" fill="none" stroke="currentColor" stroke-width="1.5"/>
    <path id="Pfad_119" data-name="Pfad 119" d="M0,0V6.895" transform="translate(297.958 1693.817) rotate(90)" fill="none" stroke="currentColor" stroke-width="1.5"/>
  </g>
</svg>
