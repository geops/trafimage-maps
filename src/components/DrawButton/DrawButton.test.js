import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import IconButton from '@material-ui/core/IconButton';
import { ReactComponent as Pencil } from '../../img/pencil.svg';
import { ReactComponent as PencilAdd } from '../../img/pencil_add.svg';
import DrawButton from '.';

describe('DrawButton', () => {
  const mockStore = configureStore([thunk]);

  const store = mockStore({
    map: {},
    app: { mapsetUrl: 'foo.mapset.ch' },
  });

  test('display basic default icon.', () => {
    const wrapper = mount(
      <Provider store={store}>
        <DrawButton />
      </Provider>,
    );
    expect(wrapper.find(Pencil).length).toBe(1);
  });

  test('display children instead of default icon.', () => {
    const wrapper = mount(
      <Provider store={store}>
        <DrawButton>
          <PencilAdd />
        </DrawButton>
      </Provider>,
    );
    expect(wrapper.find(Pencil).length).toBe(0);
    expect(wrapper.find(PencilAdd).length).toBe(1);
  });

  test('open new window to mapset with an encoded string representing the current url.', () => {
    const wrapper = mount(
      <Provider store={store}>
        <DrawButton />
      </Provider>,
    );
    global.window.open = jest.fn();
    wrapper
      .find(IconButton)
      .at(0)
      .simulate('click', { target: { name: 'width', value: 50 } });
    expect(global.window.open).toBeCalledWith(
      'foo.mapset.ch?parent=http%3A%2F%2Flocalhost%2F',
      '_self',
    );
  });
});
