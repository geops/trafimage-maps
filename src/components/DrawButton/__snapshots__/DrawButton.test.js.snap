// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawButton should match snapshot appending an encoded string to mapsetUrl. 1`] = `
<a
  aria-disabled={false}
  className="MuiButtonBase-root MuiIconButton-root"
  href="foo.mapset.ch?parent=http%3A%2F%2Flocalhost%2F"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  target="blank_"
  title="Zeichnen"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      fill="currentColor"
      focusable={false}
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12.9 6.858l4.242 4.243L7.242 21H3v-4.243l9.9-9.9zm1.414-1.414l2.121-2.122a1 1 0 0 1 1.414 0l2.829 2.829a1 1 0 0 1 0 1.414l-2.122 2.121-4.242-4.242z"
        />
      </g>
    </svg>
  </span>
</a>
`;
